src/auth/auth.guard.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { FirebaseService } from '../firebase';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(private readonly firebaseService: FirebaseService) {}
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
```

Comment:
Shouldn't it be just Promise<boolean> | Observable<boolean>

Reply:
please, see my previous comment